Kickstarter-iOS/Features/PledgePaymentMethods/Controller/NoShippingPledgeRewardsSummaryViewController.swift

```diff
@@ -0,0 +1,314 @@
+      self.tableView.leftAnchor.constraint(equalTo: self.tableViewContainer.leftAnchor),
+      self.tableView.rightAnchor.constraint(equalTo: self.tableViewContainer.rightAnchor),
+      self.tableView.topAnchor.constraint(equalTo: self.tableViewContainer.topAnchor),
+      self.tableViewContainer.leftAnchor.constraint(equalTo: self.rootStackView.leftAnchor),
+      self.tableViewContainer.rightAnchor.constraint(equalTo: self.rootStackView.rightAnchor),
+      self.tableViewContainer.topAnchor.constraint(equalTo: self.rootStackView.topAnchor),
+      self.separatorView.leftAnchor
+        .constraint(equalTo: self.rootStackView.leftAnchor, constant: Styles.grid(4)),
+      self.separatorView.rightAnchor
+        .constraint(equalTo: self.rootStackView.rightAnchor, constant: -Styles.grid(4)),
+      self.separatorView.heightAnchor.constraint(equalToConstant: 1),
+      self.rootStackView.widthAnchor.constraint(equalTo: self.view.widthAnchor)
+    ])
+  }
+
+  // MARK: - Bind Styles
+
+  override func bindStyles() {
+    super.bindStyles()
+
+    self.view.backgroundColor = .ksr_white
+    self.view.clipsToBounds = true
+
+    self.rootStackViewStyle(self.rootStackView)
+
+    self.tableViewStyle(self.tableView)
+
+    self.separatorViewStyle(self.separatorView)
+
+    self.tableViewContainerStyle(self.tableViewContainer)
+    self.tableViewContainerHeightConstraint?.constant = self.tableView.intrinsicContentSize.height
+  }
+
+  // MARK: - View model
+
+  override func bindViewModel() {
+    super.bindViewModel()
+
+    self.viewModel.outputs.loadRewardsIntoDataSource
+      .observeForUI()
+      .observeValues { [weak self] data in
+        guard let self else { return }
+
+        self.applySnapshotToDataSource(from: data)
+
+        self.setEntireViewToIsHidden(false)
+        self.tableViewContainerHeightConstraint?.constant = self.tableView.intrinsicContentSize.height
+      }
+
+    self.viewModel.outputs.configurePledgeTotalViewWithData
+      .observeForUI()
+      .observeValues { [weak self] pledgeSummaryData in
+        guard let self else { return }
+
+        self.pledgeTotalViewController.configure(with: pledgeSummaryData)
+      }
+  }
+
+  // MARK: - Configuration
+
+  func configureWith(
+    rewardsData: PostCampaignRewardsSummaryViewData,
+    bonusAmount: Double?,
+    pledgeData: PledgeSummaryViewData
+  ) {
+    self.viewModel.inputs
+      .configureWith(rewardsData: rewardsData, bonusAmount: bonusAmount, pledgeData: pledgeData)
+
+    self.view.setNeedsLayout()
+  }
+
+  // MARK: Styles
+
+  private func tableViewStyle(_ tableView: UITableView) {
+    tableView.separatorInset = .zero
+    tableView.contentInsetAdjustmentBehavior = .never
+    tableView.isScrollEnabled = false
+    tableView.delegate = self
+    tableView.rowHeight = UITableView.automaticDimension
+    tableView.separatorStyle = .none
+    tableView.backgroundColor = .ksr_white
+    tableView.translatesAutoresizingMaskIntoConstraints = false
+  }
```

Comment:
I'm setting `tableView.separatorStyle = .none` here because without it, the table view adds a separator line below each cell _and_ section header. 

I haven't found a good way to remove the lines from the section headers, yet. Here's a screenshot of what this looks like.

![simulator_screenshot_B3AAAF1E-A06F-446D-B8DE-DA16E7A9EECA](https://github.com/user-attachments/assets/81bd3afe-b138-43da-9546-a37892d0dd88)

If anyone has ideas short of adding a new separator view manually within each cell, I'd love to hear them.